{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "From last lecture we have 2 sites, one nonbinding and one binding where we take features over 240 sites. \n",
    "\n",
    "\n",
    "<img src=\"10312019_01.png\">\n",
    "Altman;s Feature program: to predict binding sites. \n",
    "build classifier to find new protein binding sites. build a grid over the protein and figute out \n",
    "the grid spacing, sufficiently finely spaced. grid 1.5A usually. at each point in protein build a feature vector\n",
    "trained on past binding sites. Most of these will look like nonsites. Can use some supervised method, cant do DL cause \n",
    "not enough data. That is how you identify 3d motifs. he does this in his lab in 1995. Steve bagly. \n",
    "\n",
    "<img src=\"10312019_02.png\">\n",
    "pocket feature: for comparing pockets in proteins. The pockets were too big to have spheres envelop them using \n",
    "    feature definition from above. Put points around pockets as features. Teh system altman built callef feature. \n",
    "    Put feature vectors around the pocket. Did not elaborate. He had 2 pockets, pocket1 has 5 feature vectors, \n",
    "    pocket 2 has 6 feature vectors. HE built algo that said are these 5 in pocket 1 more similar than 2 tahn by \n",
    "    chance. By chance you use the nonsite binding pockets or you pick from all your dara find a distribution. \n",
    "    Build several feature vectora around the pocket. Are teh 6 on left \n",
    "    more similar to the ones on right than by chance? If D1, D2, D3 bind to pocket 1 and pocket 2 is similar\n",
    "    it is a good estimate these drugs will bind to pocket 2 also. If pocket1 has list of drugs it binds, If this is drugs then this\n",
    "    is one way to do drug repurposing. find new uses for existing drugs. \n",
    "<img src=\"10312019_03.png\">\n",
    "repurposing. he wwrote paper last year where thers is a binding pocket in tublin that looks a lot like the estrodgen \n",
    "receptor. this site is also bound to by chemotherupatic agents. one way to stop cancer is to stop ability to make\n",
    "mircotubles or prevent attachemtn to this part of the tublin protein. There are anti estrogens which bind the estrogen \n",
    "receptor. tamoxifin for breast cancer turns otu it also binds tubulin. \n",
    "<img src=\"10312019_04.png\">\n",
    "\n",
    "<img src=\"10312019_05.png\">\n",
    "finding the similarities give the offtarget effects. when you find a similar protein small molecule will bind to\n",
    "or a similar binding site. How to quantify similarity? The features have different distirbutions. The simimlairy is\n",
    "not euclidean distance. You use the nonsites , 2-3 angstroms margin and then look for a big difference before saying\n",
    "there is a separation between binding and nonbinding site. Each one of teh 240 features is renormaliezed. Pocket feature is \n",
    "final example of a 3d motif. What are the other 2? Contrast these for pros and cons. we are describing a 3d motif as a \n",
    "vectof of 240 numbers. There were 3, 1) feature, 2) pocket feature, 3) fragfeature. \n",
    "\n",
    "<img src=\"10312019_06.png\">\n",
    "What is the difference between pocket feature\n",
    "and frag feature? Why is one better than the other? Frag feature, you cant find pockets similar to a pocket with \n",
    "your feature vectors. Want to bind drugs to this pocket but not similar to any known bindings. The trick is to associate 2 or 3 of these\n",
    "features together to anotehr binding site. Because there are interdependencies. YOu find that the 2 features together \n",
    "or the 2 sites together bind to a certain shape and the 3 together which define a fragment binds to that circualr structure. \n",
    "<img src=\"10312019_07.png\">\n",
    "So now I have a hypothessis that if I can find a small molecule that has those 2 shapes and are connected somehow\n",
    "that molecule can bind to this site. W/O using similarities.\n",
    "<img src=\"10312019_08.png\">\n",
    "the reason the 3d shells work so well is they wanted to dial up teh sensitiveity wo worrying about specificty. \n",
    "<img src=\"10312019_09.png\">\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"10312019_10.png\">\n",
    "<h6>Protein Structure Prediction</h6>\n",
    "how to fold long AA sequence? One way is to use physics and energy prediction. but there are drawbacks to this. We\n",
    "dont know how this really works except it happens quickly and we cant model this accurately. In PDB we have 150k structures. \n",
    "Before 1994 people predict structure based on linear sequence but was never replicated. they think they were getting \n",
    "good perormance which couldnt be replicated because of data leakage. They created CASP, 3 groups of people. 1) experimentalists\n",
    "who have 3d structure, they would hold off publication for 3-6 months and 2) predictors would try to predict what the structure\n",
    "was and 3) assessors evaluated predicdtions. did they get teh alpha helixes and beta strands correct, what was rmsd? between predicted \n",
    "and verified crystal structure? \n",
    "<img src=\"10312019_11.png\">\n",
    "example with one seauecne from PDb and one target sequence. ALignmant has GLVM and deletion and insertion. \n",
    "<img src=\"10312019_11a.png\">\n",
    "<h6>Categories in CASP</h6>\n",
    "homology modeling: if a sequence is 30-35% similar to another sequence in the PDB database w/150k entries then you \n",
    "are doing homology modeling. steps for homology modeling:\n",
    "    \n",
    "<p></p>\n",
    "1) find sequence alignment using project 1 alignment. \n",
    "<p></p>\n",
    "2) align the c-alpha between sequences, where the columns match those places have a c alpha at same place\n",
    "<p></p>\n",
    "3) columns with different AA ssame c-alpha but change the side chain. choose most likely side chain from database. \n",
    "<p></p>\n",
    "4) the gap and insertion are problems. \n",
    "<img src=\"10312019_12.png\">\n",
    "for the gap you have 3 positions in the pdb vs 2 in the gap. that is too far so\n",
    "you use energy minimizaetion. In the image below we have 2 sequences. The top one is our sequene\n",
    "we are trying to find the 3d strcuture for, teh second one is the PDB closest match with the matching\n",
    "columns. There are columns with the same AA and columns wiht different AA. With same AA we use the same\n",
    "structure in teh DB, align the c-alpha to same 3d location and use the same side chains. For different AA you have \n",
    "c-alpha in same place but you ahve to change the side chain. Take the DB side chain and replace with side chain from the \n",
    "AA of your sequence. If different side chain, when we are replacing, choose the most likely confirmation side chain from a database.\n",
    "L-V is different and needs side chain picking, you put L in teh most likely conformation position and put c-alpha in same place as V c-alpha. \n",
    "<img src=\"10312019_12a.png\">\n",
    "<img src=\"10312019_13.png\">\n",
    "use energy minimization for inertion also. if the distances are small enough then use enregy minimization. as\n",
    "the distances get bigger you get worse and worse results. Minimize Utotal. (has many components)\n",
    "<img src=\"10312019_14.png\">\n",
    "if seq is 80% or more same then result is same as if you solved experimentally. \n",
    "<img src=\"10312019_15.png\">\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"10312019_16.png\">\n",
    "2 casps ago they were predicting function using feature and pocket feature to proteins where they knew function. \n",
    "for low levels of siilarity can apply 3d motif finding on the new predicted structures. \n",
    "<img src=\"10312019_17.png\">\n",
    "second big category for casp is fold recognition. after millions of years evolution 2 sequences are less than 15% similar. \n",
    "but when you solve 3d structure they are the same! fold recogniotion is designed to find distant sequences which adopt the same fold. \n",
    "build library of backbone structures, alpha helix and betal strands, then do AA>\n",
    "<img src=\"10312019_18.png\">\n",
    "start w AA sequence, want 3d structure. first try homology modeling. Can do well w homology but need to get ~20% \n",
    "similar w homology for minimum motif finding. If cant get this then do fold prediction before abinitio. fold prediction you \n",
    "find 3d structure same but w different sequence. 2 methods. \n",
    "<img src=\"10312019_19.png\">\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<p> FR by dynamic programming.</p> \n",
    "1) map each AA pos in fold to environmental descriptor\n",
    "<img src=\"10312019_20.png\">\n",
    "what is descriptor; \n",
    "\n",
    "<img src=\"10312019_21.png\">\n",
    "<img src=\"10312019_22.png\">\n",
    "<img src=\"10312019_23.png\">\n",
    "<img src=\"10312019_24.png\">\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
