{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "<img src=\"motif1.png\">\n",
    "<img src=\"motif2.png\">\n",
    "TF is example of 1d motif. You can find these motifs from local alignment. \n",
    "<img src=\"motif3.png\">\n",
    "<img src=\"motif4.png\">\n",
    "<img src=\"motif5.png\">\n",
    "these numbers are seeing probability of seeing AA per column. column2 set .05 if you see nothing give equal\n",
    "probability over alphabet of 20aa. This is a scoring matrix for the 1d motif D*EG binding calcium sequecne/motif. \n",
    "<img src=\"motif6.png\">\n",
    "if you want to mess with sensitivity you put epsilon on PSSM where epsilon is some small constant not 0 because\n",
    "eventually you will see this. Similar to naive bayes trick where for unseen event you give small number and still\n",
    "have probability add up to 1. \n",
    "<img src=\"motif7.png\">\n",
    "in biology you know the positions are sometimes correlated position 1 and position 4. This isnt good for hmms which \n",
    "dont capture any of these relationships beyone the next state. PSSMS are above blosom62 in specifity but cant\n",
    "capture these position dependent probabilies like between position 1 and 4. \n",
    "<img src=\"motif8.png\">\n",
    "the blosum62 scoring matrix is derived by looking at BLOCKS database. BLOCKS is derived from hits from prosite.  \n",
    "<img src=\"motif9.png\">\n",
    "One way to describe a set of consensus sequences is to use a sequence logo. This shows DNA only but you \n",
    "can do this for proteins. Advantage: can see where in the sequence the strongest conservation as binding site for Tf. \n",
    "    Disadvantage: can be wrong bc it doesnt score indels and the background at 0 is equal probability of each nucleotide\n",
    "        which we know not to be true from PAM and Blosum62. \n",
    "    \n",
    "<img src=\"motif10.png\">\n",
    "the height is C(i) of the letters and indicates percent conserved across MSA. \n",
    "<img src=\"motif11.png\">\n",
    "lowest entropuy is when conserved. no change.\n",
    "the highest possible entropy is when ATCG are all equal probability which is -2. \n",
    "<img src=\"motif12.png\">\n",
    "3d motif. Advantage: detect binding, enzymatic active sites. enzymes are proteins catalyze.  Disadvantage of using sequnece logos to predict binding sites from height of logos is\n",
    "it doesnt take into account 3d informatiom. In 3d the binding constraints are the red dots binding to the Ca molecule then \n",
    "if you linearize this you will never find it bc the loops are all of variable\n",
    "<img src=\"motif13.png\">\n",
    "AN enzyme binding site is blue S which sits in pocket and catalyzes reactions. When you see a protein you have to find\n",
    "teh enzymatic binding site. \n",
    "<img src=\"motif14.png\">\n",
    "3d motifs can give clues to binding and active sites on protein. A Ca2+ binding site sits in a protein with 4 \n",
    "binding sites to a protein in a pocket. If you flatten out the sequence you get a big gap between the 6 binding sites \n",
    "in a linear representation. Bad news is we need a 3d structure and the database of 3d structures is limited; 150k in PDB. \n",
    "<img src=\"motif15.png\">\n",
    "<img src=\"motif16.png\">\n",
    "serine protease breaks peptide bonds and stable in stomach acid. when you eat steak it breaks down the steak into\n",
    "AA in digestion pricess. can also use for detergent if make stable at hightemps for protein stains\n",
    "<img src=\"motif17.png\">\n",
    "FFF: how to find 3d motif alignment. Sequence alignment, if sequence identity<120% cannot detect functional and structural\n",
    "    similarities. By chance get random match w/20AA. FFF is a supervised method where the similarity metric is range\n",
    "    overlap. IF any range of AA(HIS, SER) 2 AA overlap then this is similar and match for MSA. \n",
    "<img src=\"motif18.png\">\n",
    "<img src=\"motif18a.png\">\n",
    "FFF if you have additional AA then each triangle you have to compare. Disadvantaes: doesnt scale well to \n",
    "    higher chains. Vague on how to assign triangle or ranges. 2 ASP, 2 SER, 2 HIS, more than 2 triangles.  so this is \n",
    "    2x2x2 combinations, 8 combinations and if any is a hit then yes for serine protease. advanatage: fast,easy for simple \n",
    "        configuaitons for small proteins. When is this good for MSA? if you need to search one vs. entire database or match one to existing\n",
    "        matched set. Easier to compute, we can do that with PDB information.  gives general compare big vs small\n",
    "        and find me a similarity where there is a single range match between 3d structures. Sometimes good enough. Disadvantage:\n",
    "            can have FP if 1 range \n",
    "<img src=\"motif18b.png\">\n",
    "FFF Disadvantage; have to read teh biological literature.  This is a supervised classified method. \n",
    "<img src=\"motif19.png\">\n",
    "TESS: template search and superposition. Find one AA you know is important, like HIS. mark all AA within 18A AA of \n",
    "    interest. Make ball and get all AA in 18 angsteoms. SUpervised classification. \n",
    "<img src=\"motif20.png\">\n",
    "\n",
    "<img src=\"motif21.png\">\n",
    "tradeoff between sensitivity and specifity. make box bigger and smaller\n",
    "<img src=\"motif22.png\">\n",
    "Steve Bagley\n",
    "<img src=\"motif23.png\">\n",
    "<img src=\"motif24.png\">\n",
    "pick single point which is N in HIS and draw concentric shells. Evaluate all 40 properties in each shell. \n",
    "do you sum the atoms? yes. count. For example below shell 4. 40 properties, one maybe charge. Look at every atom\n",
    "what is charge? +1, .5, Put it in charge feature vector for shell 4. \n",
    "<img src=\"motif25.png\">\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
